truffleruby/src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -75,14 +92,29 @@ public static DynamicObject createRubyThread(RubyContext context, String info) {
                 Thread.NORM_PRIORITY,
                 context.getCoreLibrary().getNilObject(),
                 info,
-                context.getCoreLibrary().getNilObject(),
-                null);
+                context.getCoreLibrary().getNilObject());
 
         Layouts.THREAD.setFiberManagerUnsafe(object, new FiberManager(context, object)); // Because it is cyclic
 
         return object;
     }
 
+    private static final LibCSignalHandler EMPTY_HANDLER = sig -> {
+        // Empty
+    };
+
+    private static void setupSignalHandler(RubyContext context) {
+        int flags = 0; // NO SA_RESTART so we can interrupt blocking syscalls.
+        final SigAction action = context.getNativePlatform().createSigAction(EMPTY_HANDLER, flags);
+        if (!Signal.SIGVTALRM.defined()) {
+            throw new UnsupportedOperationException("SIGVTALRM not defined");
+        }
+        int result = context.getNativePlatform().getThreads().sigaction(Signal.SIGVTALRM.intValue(), action, null);
+        if (result != 0) {
+            throw new UnsupportedOperationException("sigaction() failed");
```

Comment:
I think it'd be helpful to include `errno` and, possibly, the `strerror` info in the exception message.